### ISSUE TITLE
gpu detection in resource and hwloc

### ISSUE THREAD

**Issue by trws** [flux-framework/flux-core#1490]
As part of adding the option to request GPUs, it would be good to also have support for detecting them in the resource hwloc module.  In general, this should not be a problem, cuda and OpenCL GPUs should show up as "CoProc" nodes in the hardware tree we get from hwloc.  The problem is that they don't in some cases, and unfortunately they're important cases.

Issue:
1. the hwloc library we have installed on our systems, GPU systems included, is built without cuda support
2. even with a version built with cuda support, the volta gpus on sierra and butte show up as VGA devices but not coprocessors unless we use hwloc-2 it seems, which is blocked by #1377 (I just retried this, the voltas show up correctly with a specially built hwloc-2.0 I just built on sierra)

Ideal solution: get toss/blueos to include an hwloc that includes cuda support, so we can just do this once

Possible stopgaps:
1. add something that asks cuda how many GPUs there are, possibly cross-referencing with hwloc
2. check how many files matching the pattern `/dev/nvidia[0-9]+` there are and call that the number of GPUs for the node (this would work on all of our systems right now)

**Comment by dongahn** [comment]
> Ideal solution: get toss/blueos to include an hwloc that includes cuda support, so we can just do this once

Do we have an hwloc-2.0 rpm built with cuda support publicly available?

**Comment by trws** [comment]
I should amend my earlier notes, it may not require 2, I tried an hwloc sub 2 with cuda but I just realized I’m not sure it’s most recent. It may be worth building the most recent pre 2 version with cuda and see what we get.  That would be an easier sell as a package to install I would think.


---
Sent from VMware Boxer

On April 26, 2018 at 2:03:10 PM PDT, Dong H. Ahn <notifications@github.com> wrote:

Ideal solution: get toss/blueos to include an hwloc that includes cuda support, so we can just do this once

Do we have an hwloc-2.0 rpm built with cuda support publicly available?
